internal/dast/start_local_scan.go

```diff
@@ -29,116 +29,237 @@ type StartLocalScanInput struct {
+				fmt.Sprintf("LOG_LEVEL=%s", logLevel),
+			},
+		},
+		&container.HostConfig{
+			AutoRemove:  true,
+			NetworkMode: "host",
+		},
+		nil, nil, "",
+	)
 	if err != nil {
 		logger.Error(
-			"unable to pull image from nullify platform ghrc",
+			"unable to create new docker container",
 			logger.Err(err),
 		)
-		return err
+		return nil, err
 	}
-	defer pullOut.Close()
 
-	_, err = io.Copy(os.Stdout, pullOut)
+	err = dockerclient.ContainerStart(ctx, containerResp.ID, container.StartOptions{})
 	if err != nil {
 		logger.Error(
-			"unable to copy image pull output to stdout",
+			"unable to start docker container",
 			logger.Err(err),
 		)
-		return err
+		return nil, err
 	}
 
-	containerResp, err := dockerclient.ContainerCreate(ctx, &container.Config{
-		Image: imageRef,
-		Cmd:   []string{string(requestBody)},
-	}, nil, nil, nil, "")
+	hijackedResponse, err := dockerclient.ContainerAttach(ctx, containerResp.ID, container.AttachOptions{
+		Stderr: true,
+		Stdout: true,
+		Stdin:  true,
+		Stream: true,
+	})
 	if err != nil {
 		logger.Error(
-			"unable to create new docker container",
+			"unable to attach to container",
 			logger.Err(err),
 		)
-		return err
+		return nil, err
 	}
 
-	defer func() {
-		if err = dockerclient.ContainerRemove(ctx, containerResp.ID, container.RemoveOptions{RemoveVolumes: true, RemoveLinks: false, Force: true}); err != nil {
-			logger.Error(
-				"unable to remove container",
-				logger.Err(err),
-			)
-		}
-	}()
-
-	if err = dockerclient.ContainerStart(ctx, containerResp.ID, container.StartOptions{}); err != nil {
+	_, err = hijackedResponse.Conn.Write(append(requestBody, '\n'))
+	if err != nil {
 		logger.Error(
-			"unable to start docker container",
+			"unable to write request body to container",
 			logger.Err(err),
 		)
-		return err
+		return nil, err
 	}
 
-	statusCh, errCh := dockerclient.ContainerWait(ctx, containerResp.ID, container.WaitConditionNotRunning)
-	select {
-	case err := <-errCh:
-		if err != nil {
-			logger.Error(
-				"error while waiting for container to finish scan",
-				logger.Err(err),
-			)
-			return err
-		}
-	case <-statusCh:
+	err = hijackedResponse.Conn.Close()
+	if err != nil {
+		logger.Error(
+			"unable to close connection to container",
+			logger.Err(err),
+		)
+		return nil, err
 	}
 
-	logsOut, err := dockerclient.ContainerLogs(ctx, containerResp.ID, container.LogsOptions{ShowStdout: true})
+	logsOut, err := dockerclient.ContainerLogs(ctx, containerResp.ID, container.LogsOptions{
+		ShowStdout: true,
+		ShowStderr: true,
+		Follow:     true,
+	})
 	if err != nil {
 		logger.Error(
 			"unable to create docker container logs",
 			logger.Err(err),
 		)
-		return err
+		return nil, err
 	}
 
-	_, err = stdcopy.StdCopy(os.Stdout, os.Stderr, logsOut)
-	if err != nil {
+	defer logsOut.Close()
+
+	var lastLine string
+
+	scanner := bufio.NewScanner(logsOut)
+	buf := make([]byte, 0, 64*1024)
+	scanner.Buffer(buf, 1024*1024)
+	for scanner.Scan() {
```

Comment:
Instead of this could you possible just query the owasp zap using the api similar to what occurs in active_scan.go in local dast.

Reply:
probably a good idea to get results from an API instead of stdout.
will do that later.